backend/app/ContentProviders/FourSquare.php

```diff
@@ -67,11 +69,15 @@ public function get(String $endpoint, String $method, array $parameters)
     public function createVenueFromAPIListing($object): Venue
     {
         $venue = new Venue();
-        $venue->setId($object->id);
-        $venue->setName($object->name);
-        $venue->setAddressHumanReadable($object->location->formattedAddress);
-        $venue->setLatitude($object->location->lat);
-        $venue->setLongitude($object->location->lng);
+        $this->setBasicVenueItems($object, $venue);
+        return $venue;
+    }
+
+    public function createDetailedVenueFromAPIListing($object): Venue
+    {
+        $venue = new DetailedVenue();
```

Comment:
Can you not give the object to the venue and have the venue set the items and remove all the setter functions from the class. This way all the venue logic is inside tge venue

Reply:
I chose to do it this way, such that there is a class FourSquare which will handle all FourSquare's logic. The rest of the application is in no way dependent on the FourSquare API, so if we would like to use another API instead, only the methods in FourSquare.php will need to be copied and modified for that API.
TL;DR it's because of more modularity

Reply:
Then i would opt for a abstract class for the data and four square class that extend it.

Reply:
There already is. The `ContentProvider` class, which FourSquare extends.

Reply:
Since it is not important for the API right now that this changes i will accept it, but would still like to see this changed in a new pull request